CMakeLists.txt

```diff
@@ -263,6 +263,10 @@ else()
   message("Git NOT FOUND")
 endif()
 
+if (DUCKDB_DISTRIBUTION)
+  add_definitions(-DDUCKDB_DISTRIBUTION=\"${DUCKDB_DISTRIBUTION}\")
+endif()
+
```

Comment:
Looking at this again, I would use a top-level CMake option something along the lines of:
```
option(DUCKDB_DISTRIBUTION "Codename of how the binary has been built" "unknown")

....

if (...)
     set(DUCKDB_DISTRIBUTION "dev" CACHE "hi there")
     if (....)
              set(DUCKDB_DISTRIBUTION "devdev" CACHE "hi there2")
     endif()
endif()

...

add_definitions(-DDUCKDB_DISTRIBUTION=\"${DUCKDB_DISTRIBUTION}\")

```
My CMake is hit and miss, but can help on this, see https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry.

Idea is that if stuff is provided explicitly (via Makefile / direct invocation), that value is kept, otherwise it can be set conditionally in CMake.

Reply:
This should also help with ensure it's always defined (see failures in Python or so)

Reply:
This is much nicer, thanks! (the only thing is that `option()` creates a boolean value that does not work well with strings, so I omitted that).